internal/store/leaf/fsck.go

```diff
@@ -79,13 +103,31 @@ func (s *Store) fsckLoop(ctx context.Context, path string) error {
 
 		debug.Log("[%s] Checking %s", path, name)
 
-		if err := s.fsckCheckEntry(ctx, name); err != nil {
-			return fmt.Errorf("failed to check %q: %w", name, err)
+		msg, errs, err := s.fsckCheckEntry(ctx, name)
+		if errs == errsNil {
+			ctx = ctxutil.AddToCommitMessageBody(ctx, msg)
+		} else {
+			errcoll += fmt.Errorf("failed to check %q:\n    %w\n", name, err).Error()
 		}
 	}
 
-	if err := s.fsckUpdatePublicKeys(ctx); err != nil {
-		out.Errorf(ctx, "Failed to update public keys: %s", err)
+	if errcoll != "" {
+		out.Errorf(ctx, errcoll)
+	}
+	if ctxutil.GetCommitMessageBody(ctx) == "" {
+		out.Errorf(ctx, "Nothing to commit: all secrets seemed to have failed")
```

Comment:
This branch is also executed if there is nothing to commit because nothing was updated. This shouldn't yield an error.